Reconstruction/RecCalorimeter/src/components/CaloTowerTool.cpp

```diff
@@ -38,62 +38,136 @@ StatusCode CaloTowerTool::initialize() {
 
   // check if readouts exist & retrieve PhiEta segmentations
   // if readout does not exist, reconstruction without this calorimeter part will be performed
+  std::pair<dd4hep::DDSegmentation::Segmentation*, SegmentationType> tmpPair;
   info() << "Retrieving Ecal barrel segmentation" << endmsg;
-  m_ecalBarrelSegmentation = retrieveSegmentation(m_ecalBarrelReadoutName);
+  tmpPair = retrieveSegmentation(m_ecalBarrelReadoutName);
```

Comment:
Could you add to the comments in CaloTowerTool.h, that either a Phi-Eta or Multi segmentation is required?

Reply:
sure, but in fact it is always PhiEta that is required. Just sometimes it can be 'hidden' behind the multi segmentation (but each sub-segmentation should still have phi-eta, even though it may have different cell size)